addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,124 @@
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createTsSourceFile(fileName: string): Promise<ts.SourceFile> {
+  const readFile = denodeify(fs.readFile);
+  return readFile(fileName, 'utf8')
+    .then((contents: string) => {
+      return ts.createSourceFile(fileName, contents, ts.ScriptTarget.ES6, true);
+    });
+}
+
+/**
+ * @return traverses through ts.Node and returns an array of moduleSpecifiers
+ * 
+ * @param {ts.SourceFile} node: Typescript Node of whose AST is being traversed
+ * 
+ */
+export function getImportClauses(node: ts.SourceFile): ModuleImport[] {
+  return node.statements
+    .filter(node => node.kind === ts.SyntaxKind.ImportDeclaration)  // Only Imports.
+    .map((node: ts.ImportDeclaration) => {
+      let moduleSpecifier = node.moduleSpecifier;
+      return {
+        specifierText: moduleSpecifier.getText().slice(1, -1),
+        pos: moduleSpecifier.pos,
+        end: moduleSpecifier.end
+      };
+    });
+}
+
+/** 
+ * @return a boolean value after it searches for a barrel (index.ts by convention) in a given path
+ * 
+ * @param filePath
+ * @param isDir boolean value to declare whether filePath is a directory or not. 
+ */
+export function hasIndexFile(filePath: string, isDir: boolean): Boolean {
+  let indexFile: string[];
+  if (isDir) {
+    indexFile = glob.sync(path.join(filePath, 'index.ts'), { nodir: true });
+  } else {
+    let pathDir = path.dirname(filePath);
+    indexFile = glob.sync(path.join(pathDir, 'index.ts'), { nodir: true });
+  };
+  return (indexFile.length > 0) ? true : false;
+}
+
+
+/**
+ * Return a map of all dependent file/s' path with their moduleSpecifier object
+ * (text, pos, end)
+ * 
+ * @param fileName file upon which other files depend 
+ * @param rootPath root of the project
+ * 
+ */
+export function getDependentFiles(fileName: string, rootPath: string): Promise<ModuleMap> {
+  const files = glob.sync(path.join(rootPath, '**/*.*.ts'), { nodir: true });
+  return Promise.all(files.map(file => createTsSourceFile(file)))
+    .then((tsFiles: ts.SourceFile[]) => tsFiles.map(file => getImportClauses(file)))
+    .then((moduleSpecifiers: ModuleImport[][]) => {
+      let allFiles: ModuleMap = {};
+      for (let index = 0; index < files.length; index++) {
+        // normalize path of the files for tests compatability (POSIX and Windows)
+        let sourcePath = path.normalize(files[index]);
+        allFiles[sourcePath] = moduleSpecifiers[index];
+      }
+      return allFiles;
+    })
+    .then((allFiles: ModuleMap) => {
+      let relevantFiles: ModuleMap = {};
+      Object.keys(allFiles).forEach(filePath => {
+        const tempModuleSpecifiers: ModuleImport[] = allFiles[filePath]
+          .filter(importClause => {
+            let modulePath = path.resolve(path.dirname(filePath), importClause.specifierText);
+            let resolvedFileName = path.resolve(fileName);
+            if (hasIndexFile(path.resolve(fileName), false)) {
```

Comment:
Just handle the no index case for now. The index case is considerably more complicated.